File: src/org/jetbrains/plugins/scala/configuration/ScalaLibraryType.scala
package org.jetbrains.plugins.scala
package configuration

import com.intellij.openapi.roots.libraries.{LibraryTypeService, LibraryType}
import com.intellij.openapi.roots.libraries.ui.{LibraryPropertiesEditor, LibraryEditorComponent}
import com.intellij.openapi.vfs.VirtualFile
import com.intellij.openapi.project.Project
import com.intellij.openapi.roots.ui.configuration.libraryEditor.DefaultLibraryRootsComponentDescriptor
import org.jetbrains.plugins.scala.icons.Icons
import javax.swing.JComponent

/**
 * @author Pavel Fatin
 */
class ScalaLibraryType extends LibraryType[ScalaLibraryProperties](ScalaLibraryKind) {
  def getIcon = Icons.SCALA_SDK

  def getCreateActionName = "Scala SDK"

  def createNewLibrary(parentComponent: JComponent, contextDirectory: VirtualFile, project: Project) =
    LibraryTypeService.getInstance.createLibraryFromFiles(new DefaultLibraryRootsComponentDescriptor(),
      parentComponent, contextDirectory, this, project)

  def createPropertiesEditor(editorComponent: LibraryEditorComponent[ScalaLibraryProperties]): LibraryPropertiesEditor =
    new ScalaLibraryPropertiesEditor(editorComponent)
}

object ScalaLibraryType {
  def instance: ScalaLibraryType =
    Option(LibraryType.findByKind(ScalaLibraryKind).asInstanceOf[ScalaLibraryType])
            .getOrElse(throw new NoSuchElementException("Scala library type not found"))
}

File: jps-plugin/src/org/jetbrains/jps/incremental/scala/model/ScalaLibraryType.java
package org.jetbrains.jps.incremental.scala.model;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.jps.incremental.scala.SettingsManager;
import org.jetbrains.jps.model.JpsElementChildRole;
import org.jetbrains.jps.model.library.JpsLibraryType;

/**
 * @author Pavel Fatin
 */
public class ScalaLibraryType implements JpsLibraryType<LibrarySettings> {
  private static final ScalaLibraryType myInstance = new ScalaLibraryType();

  @NotNull
  @Override
  public JpsElementChildRole<LibrarySettings> getPropertiesRole() {
    return SettingsManager.LIBRARY_SETTINGS_ROLE;
  }

  public static ScalaLibraryType getInstance() {
    return myInstance;
  }
}
